scripts/delete_resources.sh

```diff
@@ -1,149 +1,240 @@
+  echo "Total Record COUNT is: $RECORD_COUNT"
+  if [ "${RECORD_COUNT}" == 0 ]; then
+    echo "No Route53 records found for prefix $1. Nothing to delete."
   else
-    echo "No Route53 record found"
+    for i in $(echo $RECORD_VALUE | xargs -n1 echo)
+    do 
+        NAME="$(echo "$i" | cut -d "-" -f1)-r53.qa.rancher.space."
+        VALUE=$i
+        echo " NAME: $NAME"
+        echo " VALUE: $VALUE"
+        echo "
+{\"Changes\": [
+        {
+            \"Action\": \"DELETE\",
+            \"ResourceRecordSet\": {
+                \"Name\": \"${NAME}\",
+                \"Type\": \"CNAME\",
+                \"TTL\": 300,
+                \"ResourceRecords\": [
+                    {
+                        \"Value\": \"${VALUE}\"
+                    }
+                ]
+            }
+        }
+    ]
+}" > payload.json
+      cat payload.json
+      echo "Deleting Route53 record ${NAME} for prefix $1"
+      CHANGE_STATUS=$(aws route53 change-resource-record-sets --hosted-zone-id "${R53_ZONE_ID}" --change-batch file://"${PWD}"/payload.json)
+      echo "CHANGE STATUS: 
+      $CHANGE_STATUS"
+      STATUS_ID=$(echo "$CHANGE_STATUS" | grep Id: | cut -d " " -f 4)
+      echo "Status ID: $STATUS_ID"
+      #Get status from the change
+      aws route53 wait resource-record-sets-changed --id "$STATUS_ID"
+      echo "Successfully deleted Route53 record ${NAME}: status: ${STATUS_ID}"
+      exit
+    done
+    rm -rf "${PWD}"/payload.json
   fi
+}
+
+delete_all_resources () {
+  delete_ec2_instances "$1"
+  delete_db_resources "$1"
+  delete_lb_resources "$1"
+  delete_target_groups "$1"
+  delete_route_s3 "$1"
+}
+
+
+while getopts r:h OPTION
+do 
+  case "${OPTION}"
+    in
+    r) RESOURCES=${OPTARG};;
+    h|?)
+      echo "
+    Usage: 
+      $(basename "$0") [-r <resource_prefix>] [-h]
+
+      -r: resource prefix names. can be comma separated. 
+      -h: help. Prints usage example.
+      
+      Ex 1: 
+      ./delete_jenkins_resources.sh -r resourceprefix1,resourceprefix2,resourceprefix2
+      - This will delete all resources for the names resourceprefix1,resourceprefix2 and resourceprefix2
+      Ex 2:
+      ./delete_jenkins_resources.sh
+      - This will lookup the resource name from local tfvars file and delete the resources. This is interactive and will ask for confirmation before delete
+      Ex 3: 
+      ./delete_jenkins_resources.sh -h
+      - Print usage details for reference.
+      "
+      exit 1
+      ;;
+  esac
+done
+
+if [ "${RESOURCES}" = "" ]; then
+    echo "Working with local .env and .tfvars file to get resource name for deletion"
+    #Get resource name from tfvarslocal && change name to make more sense in this context
+    PRODUCT_NAME=$(grep ENV_PRODUCT <./config/.env | cut -d= -f2 | tr -d ' "')  # Split string based on delimiter =
+    if echo "${PRODUCT_NAME}" | grep -q "ENV_PRODUCT"; then
+      PRODUCT_NAME=$(echo "${PRODUCT_NAME}" | cut -d ":" -f 2)  # Split string based on delimiter :
```

Comment:
from where we are getting that its coming with a ` : `  ?

Reply:
yaml format values use a colon. if by mistake a person edits the .env file manually and puts a yaml format here instead of = it should still work. circumventing manual human errors.

Reply:
nice ok